scalalib/src/mill/scalalib/scalafmt/ScalafmtWorker.scala

```diff
@@ -1,20 +1,33 @@
 package mill.scalalib.scalafmt
 
-import mill._
-import mill.define.{Discover, ExternalModule, Worker}
-import mill.define.TaskCtx
-import org.scalafmt.interfaces.Scalafmt
+import mill.*
+import mill.define.{Discover, ExternalModule, PathRef, TaskCtx, Worker}
+import mill.scalalib._
 
 import scala.collection.mutable
 import mill.api.Result
 
-object ScalafmtWorkerModule extends ExternalModule {
-  def worker: Worker[ScalafmtWorker] = Task.Worker { new ScalafmtWorker() }
+object ScalafmtWorkerModule extends ExternalModule with JavaModule {
+
+  /**
+   * Classpath for running Palantir Java Format.
+   */
+  def scalafmtClasspath: T[Seq[PathRef]] = Task {
+    defaultResolver().classpath(
+      Seq(mvn"org.scalameta:scalafmt-dynamic_2.13:${mill.scalalib.api.Versions.scalafmtVersion}")
+    )
+  }
+
+  def scalafmtClassLoader: Worker[java.net.URLClassLoader] = Task.Worker {
```

Comment:
Nice trick to have the classloader in its own worker, since this avoids any explicit  ÀutoCloesable` handling. I think we should keep this `private` though.

Reply:
Whoops sorry I missed this, I guess we may come back to it as part of https://github.com/com-lihaoyi/mill/issues/4716 when we go around privatizing everything we can